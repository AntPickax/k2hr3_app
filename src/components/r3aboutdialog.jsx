/*
 *
 * K2HR3 Web Application
 *
 * Copyright 2017 Yahoo! Japan Corporation.
 *
 * K2HR3 is K2hdkc based Resource and Roles and policy Rules, gathers
 * common management information for the cloud.
 * K2HR3 can dynamically manage information as "who", "what", "operate".
 * These are stored as roles, resources, policies in K2hdkc, and the
 * client system can dynamically read and modify these information.
 *
 * For the full copyright and license information, please view
 * the license file that was distributed with this source code.
 *
 * AUTHOR:   Takeshi Nakatani
 * CREATE:   Mon Sep 4 2017
 * REVISION:
 *
 */

import React						from 'react';
import ReactDOM						from 'react-dom';						// eslint-disable-line no-unused-vars
import PropTypes					from 'prop-types';

import { withTheme, withStyles }	from '@material-ui/core/styles';		// decorator
import Button						from '@material-ui/core/Button';
import Dialog						from '@material-ui/core/Dialog';
import DialogTitle					from '@material-ui/core/DialogTitle';
import DialogContent				from '@material-ui/core/DialogContent';
import DialogContentText			from '@material-ui/core/DialogContentText';
import DialogActions				from '@material-ui/core/DialogActions';
import Typography					from '@material-ui/core/Typography';
import CancelIcon					from '@material-ui/icons/Cancel';

import { r3IsEmptyString }			from '../util/r3util';
import { r3AboutDialogStyles }		from './r3styles';

//
// Local variables
//
const k2hr3Title		=	'K2HR3';
const k2hr3LicenseType	=	'MIT';
const k2hr3Content		=	'K2HR3 is K2hdkc based Resource and Roles and policy Rules, gathers common' + 
							'management information for the cloud.' + 
							'K2HR3 can dynamically manage information as "who", "what", "operate".' + 
							'These are stored as roles, resources, policies in K2hdkc, and the client' +
							'system can dynamically read and modify these information.';
const k2hr3License		=	'Copyright(C) 2017 Yahoo Japan Corporation.';

//
// User Data(with role token) Information Class
//
@withTheme
@withStyles(r3AboutDialogStyles)
export default class R3AboutDialog extends React.Component
{
	static contextTypes = {
		r3Context:		PropTypes.object.isRequired
	};

	static propTypes = {
		r3provider:		PropTypes.object.isRequired,
		open:			PropTypes.bool.isRequired,
		onClose:		PropTypes.func.isRequired,

		licensePackage:	PropTypes.string,
		licenseType:	PropTypes.string,
		licenseText:	PropTypes.string
	};

	static defaultProps = {
		licensePackage:	null,
		licenseType:	null,
		licenseText:	null
	};

	constructor(props)
	{
		super(props);

		// Binding(do not define handlers as arrow functions for performance)
		this.handleClose	= this.handleClose.bind(this);
	}

	handleClose(event, reason)
	{
		this.props.onClose(event, reason);
	}

	getHtmlLicenseText()
	{

		// Output by <p> tag per line
		//
		let	lines = this.props.licenseText.split('\n');

		return (
			lines.map( (item, pos) => {						// eslint-disable-line no-unused-vars
				return (
					<Typography key={ pos }>
						{ item }
						<br />
					</Typography>
				);
			})
		);
	}

	getLicenseType()
	{
		return (
			<Typography { ...this.props.theme.r3AboutDialog.licenseType }>
				License: { (r3IsEmptyString(this.props.licensePackage) ? k2hr3LicenseType : this.props.licenseType) }
				<br />
			</Typography>
		);
	}

	getContentText()
	{
		if(r3IsEmptyString(this.props.licensePackage)){
			return (
				<Typography { ...this.props.theme.r3AboutDialog.content }>
					{ k2hr3Content }
					<br />
					<br />
					{ k2hr3License }
				</Typography>
			);
		}else{
			return (
				<Typography { ...this.props.theme.r3AboutDialog.content }>
					{ this.getHtmlLicenseText() }
				</Typography>
			);
		}
	}

	render()
	{
		const { theme, classes, r3provider } = this.props;

		let	licenseType	= this.getLicenseType();
		let	contentText	= this.getContentText();

		return (
			<Dialog
				open={ this.props.open }
				onClose={ (event, reason) => this.handleClose(event, reason) }
				{ ...theme.r3AboutDialog.root }
				className={ classes.root }
			>
				<DialogTitle
					{ ...theme.r3AboutDialog.dialogTitle }
					className={ classes.dialogTitle }
				>
					<Typography
						{ ...theme.r3AboutDialog.title }
						className={ classes.title }
					>
						About { (r3IsEmptyString(this.props.licensePackage) ? k2hr3Title : this.props.licensePackage) }
					</Typography>
				</DialogTitle>
				<DialogContent 
					className={ classes.dialogContent }
				>
					<DialogContentText
						{ ...theme.r3AboutDialog.dialogContentText }
						className={ classes.dialogContentText }
					>
						{ licenseType }
						{ contentText }
					</DialogContentText>
				</DialogContent>
				<DialogActions
					className={ classes.dialogAction }
				>
					<Button
						onClick={ (event) => this.handleClose(event, null) }
						{ ...theme.r3AboutDialog.button }
						className={ classes.button }
					>
						{ r3provider.getR3TextRes().tResButtonClose }
						<CancelIcon
							className={ classes.buttonIcon }
						/>
					</Button>
				</DialogActions>
			</Dialog>
		);
	}
}

/*
 * VIM modelines
 *
 * vim:set ts=4 fenc=utf-8:
 *
 */
