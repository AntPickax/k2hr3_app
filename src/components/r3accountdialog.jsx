/*
 *
 * K2HR3 Web Application
 *
 * Copyright 2017 Yahoo Japan Corporation.
 *
 * K2HR3 is K2hdkc based Resource and Roles and policy Rules, gathers
 * common management information for the cloud.
 * K2HR3 can dynamically manage information as "who", "what", "operate".
 * These are stored as roles, resources, policies in K2hdkc, and the
 * client system can dynamically read and modify these information.
 *
 * For the full copyright and license information, please view
 * the license file that was distributed with this source code.
 *
 * AUTHOR:   Takeshi Nakatani
 * CREATE:   Fri Sep 24 2021
 * REVISION:
 *
 */

import React						from 'react';
import ReactDOM						from 'react-dom';						// eslint-disable-line no-unused-vars
import PropTypes					from 'prop-types';

import Button						from '@mui/material/Button';
import Dialog						from '@mui/material/Dialog';
import DialogTitle					from '@mui/material/DialogTitle';
import DialogContent				from '@mui/material/DialogContent';
import DialogActions				from '@mui/material/DialogActions';
import Typography					from '@mui/material/Typography';
import Tooltip						from '@mui/material/Tooltip';
import TextField					from '@mui/material/TextField';
import CheckCircleIcon				from '@mui/icons-material/CheckCircle';
import CopyClipBoardIcon			from '@mui/icons-material/AssignmentTurnedInRounded';

import { r3AccountDialog }			from './r3styles';
import { r3IsEmptyString, r3IsEmptyEntityObject, r3IsSafeTypedEntity } from '../util/r3util';

const unscopedtokenFieldName	= 'unscopedtoken-textfield';

//
// Create New Path Dialog Class
//
export default class R3AccountDialog extends React.Component
{
	static propTypes = {
		r3provider:		PropTypes.object.isRequired,
		open:			PropTypes.bool,
		username:		PropTypes.string,
		unscopedtoken:	PropTypes.string,
		onClose:		PropTypes.func.isRequired
	};

	static defaultProps = {
		open:			false,
		username:		null,
		unscopedtoken:	null
	};

	unscopedtokenInputElement = null;												// unscopedtoken textfield for registration code.

	state = {
		open:		this.props.open,
		tooltips: {
			copyClipboardButtonTooltip:		false
		}
	};

	constructor(props)
	{
		super(props);

		this.handleCopyClipboard = this.handleCopyClipboard.bind(this);

		// styles
		this.sxClasses = r3AccountDialog(props.theme);
	}

	// [NOTE]
	// Use getDerivedStateFromProps by deprecating componentWillReceiveProps in React 17.x.
	// The only purpose is to set the state data from props when the dialog changes from hidden to visible.
	//
	static getDerivedStateFromProps(nextProps, prevState)
	{
		if(prevState.open != nextProps.open){
			if(nextProps.open){
				// Inivisible to Visible
				return {
					open:	nextProps.open
				};
			}else{
				// Visible to Inivisible
				return {
					open:	nextProps.open
				};
			}
		}
		return null;														// Return null to indicate no change to state.
	}

	handleCopyClipboardButtonTooltipChange = (event, isOpen) =>				// eslint-disable-line no-unused-vars
	{
		this.setState({
			tooltips: {
				copyClipboardButtonTooltip:	isOpen
			}
		});
	};

	handleCopyClipboard(event)												// eslint-disable-line no-unused-vars
	{
		if(r3IsEmptyEntityObject(this.unscopedtokenInputElement, 'select') || !r3IsSafeTypedEntity(this.unscopedtokenInputElement.select, 'function')){
			return;
		}
		this.unscopedtokenInputElement.select();	// select all text in text field
		document.execCommand('copy');				// cpoy to clipboard
		window.getSelection().removeAllRanges();	// unselect text
		this.unscopedtokenInputElement.blur();		// off furcus

		this.setState({
			tooltips: {
				copyClipboardButtonTooltip:	false
			}
		});
	}

	render()
	{
		const { theme, r3provider } = this.props;

		let	username;
		let	usernameClass;
		if(!r3IsEmptyString(this.props.username)){
			username		= this.props.username;
			usernameClass	= this.sxClasses.value;
		}else{
			username		= r3provider.getR3TextRes().tResUnknownUsernameLabel;
			usernameClass	= this.sxClasses.valueItalic;
		}

		let	unscopedtoken;
		if(!r3IsEmptyString(this.props.unscopedtoken)){
			unscopedtoken	= this.props.unscopedtoken;
		}else{
			unscopedtoken	= r3provider.getR3TextRes().tResNoUnscopedTokenLabel;
		}

		return (
			<Dialog
				open={ this.props.open }
				onClose={ (event, reason) => this.props.onClose(event, reason) }
				{ ...theme.r3AccountDialog.root }
				sx={ this.sxClasses.root }
			>
				<DialogTitle
					{ ...theme.r3AccountDialog.dialogTitle }
					sx={ this.sxClasses.dialogTitle }
				>
					<Typography
						{ ...theme.r3AccountDialog.title }
						sx={ this.sxClasses.title }
					>
						{ r3provider.getR3TextRes().cAccountTitle }
					</Typography>
				</DialogTitle>

				<DialogContent
					sx={ this.sxClasses.dialogContent }
				>
					<Typography
						{ ...theme.r3AccountDialog.subTitle }
						sx={ this.sxClasses.subTitle }
					>
						{ r3provider.getR3TextRes().tResAccoutUsernameTitle }
					</Typography>
					<Typography
						{ ...theme.r3AccountDialog.value }
						sx={ usernameClass }
					>
						{ username }
					</Typography>
					<Typography
						{ ...theme.r3AccountDialog.subTitle }
						sx={ this.sxClasses.subTitle }
					>
						{ r3provider.getR3TextRes().tResUnscopedTokenTitle }
					</Typography>
					<TextField
						name={ unscopedtokenFieldName }
						value={ unscopedtoken }
						inputRef = { (element) => { this.unscopedtokenInputElement = element; } } 
						InputProps={{ sx: this.sxClasses.unscopedtokenInputTextField }}
						inputProps={{ style: { padding: 0 } }}
						{ ...theme.r3AccountDialog.unscopedtokenTextField }
						sx={ this.sxClasses.unscopedtokenTextField }
					/>
					<Tooltip
						title={ r3provider.getR3TextRes().tResCopyClipboardTT }
						open={ ((r3IsEmptyEntityObject(this.state, 'tooltips') || !r3IsSafeTypedEntity(this.state.tooltips.copyClipboardButtonTooltip, 'boolean')) ? false : this.state.tooltips.copyClipboardButtonTooltip) }
					>
						<Button
							onClick={ this.handleCopyClipboard }
							onMouseEnter={ event => this.handleCopyClipboardButtonTooltipChange(event, true) }
							onMouseLeave={ event => this.handleCopyClipboardButtonTooltipChange(event, false) }
							{ ...theme.r3AccountDialog.copyClipboardButton }
							sx={ this.sxClasses.copyClipboardButton }
						>
							<CopyClipBoardIcon
								sx={ this.sxClasses.copyClipboardIcon }
							/>
							{ r3provider.getR3TextRes().tResCopyClipboardButton }
						</Button>
					</Tooltip>
				</DialogContent>
				<DialogActions>
					<Button
						disabled={ false }
						onClick={ (event) => this.props.onClose(event, null) }
						{ ...theme.r3AccountDialog.okButton }
						sx={ this.sxClasses.okButton }
					>
						{ r3provider.getR3TextRes().tResButtonOk }
						<CheckCircleIcon
							sx={ this.sxClasses.buttonIcon }
						/>
					</Button>
				</DialogActions>
			</Dialog>
		);
	}
}

/*
 * Local variables:
 * tab-width: 4
 * c-basic-offset: 4
 * End:
 * vim600: noexpandtab sw=4 ts=4 fdm=marker
 * vim<600: noexpandtab sw=4 ts=4
 */
