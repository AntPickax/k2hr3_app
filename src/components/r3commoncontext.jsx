/*
 *
 * K2HR3 Web Application
 *
 * Copyright 2022 Yahoo! Japan Corporation.
 *
 * K2HR3 is K2hdkc based Resource and Roles and policy Rules, gathers
 * common management information for the cloud.
 * K2HR3 can dynamically manage information as "who", "what", "operate".
 * These are stored as roles, resources, policies in K2hdkc, and the
 * client system can dynamically read and modify these information.
 *
 * For the full copyright and license information, please view
 * the license file that was distributed with this source code.
 *
 * AUTHOR:   Takeshi Nakatani
 * CREATE:   Tue Jul 5 2022
 * REVISION:
 *
 */

import React from 'react';

//
// Default(empty) common context for all components
//
// [NOTE]
// React 18 (MUI v5) can not use withTheme HOC function, and has to use
// React.createContext for creating common(global) context.
//
// [TODO]
// Currently, only common information data(r3Context) is set in this
// R3CommonContext, but it should be better to include Theme and Provider
// in the future.
// Since Theme and Provider are also transferred from the upper component
// to the child component, it seems appropriate to include them in this
// R3CommonContext.
// However, React and MUI are subject to frequent specification changes,
// so we will consider this after it stabilizes.
//
export const R3CommonContext = React.createContext({r3Context: {}});

/*
 * VIM modelines
 *
 * vim:set ts=4 fenc=utf-8:
 *
 */
