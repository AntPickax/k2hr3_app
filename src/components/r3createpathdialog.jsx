/*
 *
 * K2HR3 Web Application
 *
 * Copyright 2017 Yahoo! Japan Corporation.
 *
 * K2HR3 is K2hdkc based Resource and Roles and policy Rules, gathers
 * common management information for the cloud.
 * K2HR3 can dynamically manage information as "who", "what", "operate".
 * These are stored as roles, resources, policies in K2hdkc, and the
 * client system can dynamically read and modify these information.
 *
 * For the full copyright and license information, please view
 * the license file that was distributed with this source code.
 *
 * AUTHOR:   Takeshi Nakatani
 * CREATE:   Mon Sep 4 2017
 * REVISION:
 *
 */

import React			from 'react';
import ReactDOM			from 'react-dom';									// eslint-disable-line no-unused-vars
import PropTypes		from 'prop-types';

import TextField		from 'material-ui/TextField';
import FontIcon			from 'material-ui/FontIcon';
import RaisedButton		from 'material-ui/RaisedButton';
import Dialog			from 'material-ui/Dialog';

import { r3IsEmptyStringObject, r3IsEmptyString } from '../util/r3util';

//
// Create New Path Dialog Class
//
export default class R3CreatePathDialog extends React.Component
{
	constructor(props)
	{
		super(props);

		this.state = {
			newPath:	this.props.newPath
		};

		this.handleNewPathChange	= this.handleNewPathChange.bind(this);
	}

	componentWillReceiveProps(nextProps)										// eslint-disable-line no-unused-vars
	{
		this.setState({
			newPath:	nextProps.newPath
		});
	}

	handleNewPathChange(event, changedValue)									// eslint-disable-line no-unused-vars
	{
		this.setState({
			newPath:	changedValue
		});
	}

	render()
	{
		let	tenant		= r3IsEmptyStringObject(this.props.tenant, 'display') ?	'(Unselected)' : this.props.tenant.display;
		let	type		= r3IsEmptyString(this.props.type) ?					'(Unselected)' : this.props.type;
		let	parentPath	= r3IsEmptyString(this.props.parentPath) ?				'(Unselected)' : this.props.parentPath;
		let	actions		= [
			<RaisedButton
				label={ 'CANCEL' }
				labelPosition={ 'after' }
				secondary={ true }
				style={ this.context.muiTheme.r3FormButtons.raisedButtonStyle }
				disabled={ false }
				onClick={ (event) => this.props.onClose(event, false, null) }
				icon={
					<FontIcon className={ this.context.muiTheme.r3IconFonts.cancelIconFont } />
				}
			/>,
			<RaisedButton
				label={ 'OK' }
				labelPosition={ 'after' }
				primary={ true }
				style={ this.context.muiTheme.r3FormButtons.raisedButtonStyle }
				disabled={ ('' === this.state.newPath) }
				onClick={ (event) => this.props.onClose(event, true, this.state.newPath) }
				icon={
					<FontIcon className={ this.context.muiTheme.r3IconFonts.checkIconFont } />
				}
			/>
		];

		return (
			<Dialog
				title={ this.props.r3provider.getR3TextRes().cCreateNewPathTitle }
				actions={ actions }
				modal={ true }
				open={ this.props.open }
				onRequestClose={ (event) => this.props.onClose(event, false, null) }
			>
				<p style={ this.context.muiTheme.dialogSimple.keyTitleStyle } >TENANT</p>
				<p style={ this.context.muiTheme.dialogSimple.valueStyle } >{ tenant }</p>
				<p style={ this.context.muiTheme.dialogSimple.keyTitleStyle } >TYPE</p>
				<p style={ this.context.muiTheme.dialogSimple.valueStyle } >{ type }</p>
				<p style={ this.context.muiTheme.dialogSimple.keyTitleStyle } >PARENT PATH</p>
				<p style={ this.context.muiTheme.dialogSimple.valueStyle } >{ parentPath }</p>
				<p style={ this.context.muiTheme.dialogSimple.keyTitleStyle } >CREATE PATH</p>
				<TextField
					name={ 'new_path' }
					value={ this.state.newPath }
					hintText={ 'Input create path'  }
					style={ this.context.muiTheme.dialogSimple.TextFieldStyle }
					onChange={ (event, value) => this.handleNewPathChange(event, value) }
				/>
			</Dialog>
		);

	}
}

R3CreatePathDialog.contextTypes = {
	muiTheme:		PropTypes.object.isRequired,
	r3Context:		PropTypes.object.isRequired
};

R3CreatePathDialog.propTypes = {
	r3provider:		PropTypes.object.isRequired,
	open:			PropTypes.bool,
	tenant:			PropTypes.object,
	type:			PropTypes.string,
	parentPath:		PropTypes.string,
	newPath:		PropTypes.string,

	onClose:		PropTypes.func.isRequired
};

R3CreatePathDialog.defaultProps = {
	open:			false,
	tenant:			null,
	type:			null,
	parentPath:		null,
	newPath:		'',
};

/*
 * VIM modelines
 *
 * vim:set ts=4 fenc=utf-8:
 *
 */
