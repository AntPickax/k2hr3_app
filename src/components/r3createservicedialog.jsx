/*
 *
 * K2HR3 Web Application
 *
 * Copyright 2017 Yahoo! Japan Corporation.
 *
 * K2HR3 is K2hdkc based Resource and Roles and policy Rules, gathers
 * common management information for the cloud.
 * K2HR3 can dynamically manage information as "who", "what", "operate".
 * These are stored as roles, resources, policies in K2hdkc, and the
 * client system can dynamically read and modify these information.
 *
 * For the full copyright and license information, please view
 * the license file that was distributed with this source code.
 *
 * AUTHOR:   Takeshi Nakatani
 * CREATE:   Fri Nov 24 2017
 * REVISION:
 *
 */

import React			from 'react';
import ReactDOM			from 'react-dom';									// eslint-disable-line no-unused-vars
import PropTypes		from 'prop-types';

import TextField		from 'material-ui/TextField';
import FontIcon			from 'material-ui/FontIcon';
import RaisedButton		from 'material-ui/RaisedButton';
import Dialog			from 'material-ui/Dialog';

import { r3IsEmptyStringObject } from '../util/r3util';

//
// Create New Path Dialog Class
//
export default class R3CreateServiceDialog extends React.Component
{
	constructor(props)
	{
		super(props);

		this.state = {
			newServiceName:	this.props.newServiceName,
			newVerify:		this.props.newVerify
		};

		this.handleNewServiceNameChange	= this.handleNewServiceNameChange.bind(this);
		this.handleNewVerifyChange		= this.handleNewVerifyChange.bind(this);
	}

	componentWillReceiveProps(nextProps)										// eslint-disable-line no-unused-vars
	{
		this.setState({
			newServiceName:	nextProps.newServiceName,
			newVerify:		nextProps.newVerify
		});
	}

	handleNewServiceNameChange(event, changedValue)								// eslint-disable-line no-unused-vars
	{
		this.setState({
			newServiceName:	changedValue
		});
	}

	handleNewVerifyChange(event, changedValue)									// eslint-disable-line no-unused-vars
	{
		this.setState({
			newVerify:	changedValue
		});
	}

	render()
	{
		let	tenant		= r3IsEmptyStringObject(this.props.tenant, 'display') ? '(Unselected)' : this.props.tenant.display;
		let	actions		= [
			<RaisedButton
				label={ 'CANCEL' }
				labelPosition={ 'after' }
				secondary={ true }
				style={ this.context.muiTheme.r3FormButtons.raisedButtonStyle }
				disabled={ false }
				onClick={ (event) => this.props.onClose(event, false, null, null) }
				icon={
					<FontIcon className={ this.context.muiTheme.r3IconFonts.cancelIconFont } />
				}
			/>,
			<RaisedButton
				label={ 'OK' }
				labelPosition={ 'after' }
				primary={ true }
				style={ this.context.muiTheme.r3FormButtons.raisedButtonStyle }
				disabled={ ('' === this.state.newServiceName) }
				onClick={ (event) => this.props.onClose(event, true, this.state.newServiceName, this.state.newVerify) }
				icon={
					<FontIcon className={ this.context.muiTheme.r3IconFonts.checkIconFont } />
				}
			/>
		];

		return (
			<Dialog
				title={ this.props.r3provider.getR3TextRes().cCreateServiceTitle }
				actions={ actions }
				modal={ true }
				open={ this.props.open }
				onRequestClose={ (event) => this.props.onClose(event, false, null, null) }
			>
				<p style={ this.context.muiTheme.dialogSimple.keyTitleStyle } >TENANT for SERVICE owner</p>
				<p style={ this.context.muiTheme.dialogSimple.valueStyle } >{ tenant }</p>
				<p style={ this.context.muiTheme.dialogSimple.keyTitleStyle } >CREATE SERVICE</p>
				<TextField
					name={ 'new_service_name' }
					value={ this.state.newServiceName }
					hintText={ 'Input create service name'  }
					style={ this.context.muiTheme.dialogSimple.TextFieldStyle }
					onChange={ (event, value) => this.handleNewServiceNameChange(event, value) }
				/>
				<p style={ this.context.muiTheme.dialogSimple.keyTitleStyle } >VERIFY URL or STATIC RESOURCE</p>
				<TextField
					name={ 'new_verify' }
					value={ this.state.newVerify }
					hintText={ 'Input verify'  }
					style={ this.context.muiTheme.dialogSimple.TextFieldStyle }
					onChange={ (event, value) => this.handleNewVerifyChange(event, value) }
				/>
			</Dialog>
		);

	}
}

R3CreateServiceDialog.contextTypes = {
	muiTheme:		PropTypes.object.isRequired,
	r3Context:		PropTypes.object.isRequired
};

R3CreateServiceDialog.propTypes = {
	r3provider:		PropTypes.object.isRequired,
	open:			PropTypes.bool,
	tenant:			PropTypes.object,
	newServiceName:	PropTypes.string,
	newVerify:		PropTypes.string,

	onClose:		PropTypes.func.isRequired
};

R3CreateServiceDialog.defaultProps = {
	open:			false,
	tenant:			null,
	newServiceName:	'',
	newVerify:		''
};

/*
 * VIM modelines
 *
 * vim:set ts=4 fenc=utf-8:
 *
 */
