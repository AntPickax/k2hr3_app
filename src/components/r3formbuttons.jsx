/*
 *
 * K2HR3 Web Application
 *
 * Copyright 2017 Yahoo! Japan Corporation.
 *
 * K2HR3 is K2hdkc based Resource and Roles and policy Rules, gathers
 * common management information for the cloud.
 * K2HR3 can dynamically manage information as "who", "what", "operate".
 * These are stored as roles, resources, policies in K2hdkc, and the
 * client system can dynamically read and modify these information.
 *
 * For the full copyright and license information, please view
 * the license file that was distributed with this source code.
 *
 * AUTHOR:   Takeshi Nakatani
 * CREATE:   Thu Aug 24 2017
 * REVISION:
 *
 */

import React						from 'react';
import ReactDOM						from 'react-dom';						// eslint-disable-line no-unused-vars
import PropTypes					from 'prop-types';

import { withTheme, withStyles }	from '@material-ui/core/styles';		// decorator
import Button						from '@material-ui/core/Button';
import CheckCircleIcon				from '@material-ui/icons/CheckCircle';
import CancelIcon					from '@material-ui/icons/Cancel';

import { r3FormButtons }			from './r3styles';

//
// Form Button Class
//
@withTheme
@withStyles(r3FormButtons)
export default class R3FormButtons extends React.Component
{
	static contextTypes = {
		r3Context:	PropTypes.object.isRequired
	};

	static propTypes = {
		r3provider:	PropTypes.object.isRequired,
		status:		PropTypes.bool,
		onSave:		PropTypes.func.isRequired,
		onCancel:	PropTypes.func.isRequired
	};

	static defaultProps = {
		status:		false
	};

	constructor(props)
	{
		super(props);
	}

	render()
	{
		const { theme, classes, r3provider } = this.props;

		return (
			<div
				className={ classes.root }
			>
				<Button
					disabled={ !this.props.status }
					onClick={ (event) => this.props.onCancel(event) }
					{ ...theme.r3FormButtons.cancelButton }
					className={ classes.cancelButton }
				>
					{ r3provider.getR3TextRes().tResButtonCancel }
					<CancelIcon
						className={ classes.buttonIcon }
					/>
				</Button>
				<Button
					disabled={ !this.props.status }
					onClick={ (event) => this.props.onSave(event) }
					{ ...theme.r3FormButtons.saveButton }
					className={ classes.saveButton }
				>
					{ r3provider.getR3TextRes().tResButtonSave }
					<CheckCircleIcon
						className={ classes.buttonIcon }
					/>
				</Button>
			</div>
		);
	}
}

/*
 * VIM modelines
 *
 * vim:set ts=4 fenc=utf-8:
 *
 */
