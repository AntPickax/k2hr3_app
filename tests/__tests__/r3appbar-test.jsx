/*
 *
 * K2HR3 Web Application
 *
 * Copyright 2017 Yahoo Japan Corporation.
 *
 * K2HR3 is K2hdkc based Resource and Roles and policy Rules, gathers
 * common management information for the cloud.
 * K2HR3 can dynamically manage information as "who", "what", "operate".
 * These are stored as roles, resources, policies in K2hdkc, and the
 * client system can dynamically read and modify these information.
 *
 * For the full copyright and license information, please view
 * the license file that was distributed with this source code.
 *
 * AUTHOR:   Takeshi Nakatani
 * CREATE:   Mon Jul 9 2018
 * REVISION:
 *
 */

import React					from 'react';										// eslint-disable-line no-unused-vars
import renderer					from 'react-test-renderer';
import getElementWithContext	from 'react-test-context-provider';					// for context provider
import { ThemeProvider }		from '@mui/material/styles';
import { StyledEngineProvider, CssBaseline}	from '@mui/material';					// for jss and reset.css

import r3Theme					from '../../src/components/r3theme';				// custom theme
import R3AppBar					from '../../src/components/r3appbar';
import { R3CommonContext }		from '../../src/components/r3commoncontext';
import R3Provider				from '../../src/util/r3provider';

import mock_fetch				from '../__mocks__/fetchMock';						// eslint-disable-line no-unused-vars
import { createNodeMock }		from '../__mocks__/materialUiMock';					// for material-ui

//
// Mock functions
//
// [NOTE]
// If you need to customize return value, you can call methods
// for return value by each mock function.
// see: https://jestjs.io/docs/ja/mock-functions
//
// ex)	mockfunc
//			.mockReturnValueOnce(10)
//			.mockReturnValueOnce('x')
//			.mockReturnValue(true);
//
const treedetach	= jest.fn();										// eslint-disable-line no-undef
const opentree		= jest.fn();										// eslint-disable-line no-undef
const checkupdating	= jest.fn();										// eslint-disable-line no-undef
const about			= jest.fn();										// eslint-disable-line no-undef
const sign			= jest.fn();										// eslint-disable-line no-undef
const account		= jest.fn();										// eslint-disable-line no-undef

//
// Main test
//
describe('R3AppBar', () => {											// eslint-disable-line no-undef
	it('test snapshot for R3AppBar', () => {							// eslint-disable-line no-undef
		const r3provider	= new R3Provider(null);
		const commonContext	= { r3Context: r3provider.getR3Context() };

		const element		= getElementWithContext(
			{
				r3Context:	r3provider.getR3Context()
			},
			<StyledEngineProvider injectFirst>
				<ThemeProvider theme={ r3Theme } >
					<CssBaseline />
					<R3CommonContext.Provider
						value={ commonContext }
					>
						<R3AppBar
							theme={ r3Theme }
							r3provider={ r3provider }
							title='K2HR3'
							enDock={ false }
							isDocking={ true }
							onTreeDetach={ treedetach }
							onOpenTree={ opentree }
							onCheckUpdating={ checkupdating }
							onAbout={ about }
							onSign={ sign }
							onAccount={ account }
						/>
					</R3CommonContext.Provider>
				</ThemeProvider>
			</StyledEngineProvider>
		);

		const component = renderer.create(element, { createNodeMock });
		let tree		= component.toJSON();
		expect(tree).toMatchSnapshot();									// eslint-disable-line no-undef
	});
});

/*
 * Local variables:
 * tab-width: 4
 * c-basic-offset: 4
 * End:
 * vim600: noexpandtab sw=4 ts=4 fdm=marker
 * vim<600: noexpandtab sw=4 ts=4
 */
