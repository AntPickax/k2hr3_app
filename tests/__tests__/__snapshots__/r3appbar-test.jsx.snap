// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`R3AppBar test snapshot for R3AppBar 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-5k0x2i-MuiPaper-root-MuiAppBar-root"
  style={
    {
      "--Paper-shadow": "0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12)",
    }
  }
>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-s3ewxw-MuiToolbar-root"
  >
    <button
      aria-haspopup="true"
      aria-label="main menu"
      aria-labelledby={null}
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-nxvj6i-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element={true}
      disabled={null}
      label="main-menu"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14500qs-MuiSvgIcon-root"
        data-testid="MenuIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
        />
      </svg>
    </button>
    <h6
      className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-3vpxrl-MuiTypography-root"
      style={{}}
    >
      K2HR3
    </h6>
    <button
      aria-haspopup="true"
      aria-label="signin menu"
      aria-labelledby={null}
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-nxvj6i-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element={true}
      disabled={null}
      label="signin-menu"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14500qs-MuiSvgIcon-root"
        data-testid="AccountCircleIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6m0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20"
        />
      </svg>
    </button>
  </div>
</header>
`;
